B2B phase 2 changes (#8546)
auth0/docs: articles/architecture-scenarios/_includes/_logout/_introduction.md

* Which sessions should be terminated when the user initiates logout?
* What information should you provide to users as confirmation of the sessions terminated?
* Where should users be redirected to after logout completes?
* How long do you want sessions to last in the event that users do not trigger the logout process?
<% if (platform === "b2b") { %>

> * Should the End User be logged out of all of their application sessions when they log out of one of their applications?

Reviewer: ```suggestion
* Should the End User be logged out of all of their application sessions when they log out of one?
```

> * Should the session at the organization's IDP also be terminated at logout?

Reviewer: ```suggestion
* Should the session with an organization's IDP also be terminated at logout?
```

<% } %>

> Given the varying types of sessions that can be created whenever a user logs in, there are several types of logout possible. Local application logout ends the session with the application, whereas Auth0 logout [terminates the Auth0 session](/logout/guides/logout-auth0). [Federated logout](/logout/guides/logout-idps) terminates the Auth0 session and also propagates the logout request to a remote, third-party Identity Provider. Global, or [Single Logout](/logout/guides/logout-applications) (SLO), ends the Auth0 session and also sends a logout request/notice to applications relying on the Auth0 session.

Reviewer: Hmmm. We talk about federated logout here. Is the additional paragraph below (https://github.com/auth0/docs/pull/8546/files#r356571991) necessary?

Author: That's a good question... maybe we should just point this link to the federated logout section...

Author: ```suggestion
Given the varying types of sessions that can be created whenever a user logs in, there are several types of logout possible. Local application logout ends the session with the application, whereas Auth0 logout [terminates the Auth0 session](/logout/guides/logout-auth0). If you have organizations that are using their own IDP, you may want to consider a [Federated Logout](#federated-logout) strategy and implement accordingly. Global, or [Single Logout](/logout/guides/logout-applications) (SLO), ends the Auth0 session and also sends a logout request/notice to applications relying on the Auth0 session.
```

If the logout feature in one application terminates an Auth0 SSO session that is used by other applications, the user may lose work if they have uncommitted transactions. Be sure to add the functionality needed to handle such conditions to minimize the likelihood of lost work.
:::

<% if (platform === "b2b") { %>

> In some situations, a user may expect to be logged out of all of their applications when they log out of a single application in your system.  This is something that can add complexity to a system, but if you have sensitive data and are concerned that a user may leave themselves open after logging out because they didn't realize they had to log out of all of their applications, you will need to review [Single Logout](#single-logout).

Reviewer: ```suggestion
In some situations, a user may be expected to logout of all associated applications when they log out of any one of the applications you provide. This is something that can add complexity. However if you have concerns that users could leave themselves vulnerable (perhaps due to data sensitivity or the like), then you will likely need to review [Single Logout](#single-logout) and implement accordingly.
```

> If you have organizations that are using their own IDP, you may want to consider [Federated Logout](#federated-logout)

Reviewer: Hmmm. We talk about Federated logout above (https://github.com/auth0/docs/pull/8546/files#r356572969). Are both of these paragraphs necessary? 

```suggestion
If you have organizations that are using their own IDP, you may want to consider a [Federated Logout](#federated-logout) strategy and implement accordingly.
```

Author: ```suggestion
```

Not all users will trigger the logout process manually, so Auth0 also provides **session timeout** to prevent overly long-lived sessions. This setting is [available and configurable via the Auth0 Dashboard](/dashboard/reference/settings-tenant#login-session-management).

<% if (platform === "b2b") { %>

>  [Federated User Logout](/logout/guides/logout-idps) may be something that you need to consider for your application.  If you or your customers will be using a third-party IdP (i.e., something other than a [Database Identity Provider](/connections/database)) then the question of whether you need to log the user out of the IdP when they log out of your application is something you will need to answer. The answer depends on what your users would expect. If the application and/or IdP you use is tied closely to a customer organization and a central part of day-to-day operations, then it may be frustrating for users to get logged out of their IdP when they log out of your application. If not, then being logged out of the IdP may be expected, or in some cases even desired. In most B2B scenarios, our customers find that it is preferable *not* to perform federated logout for a user.

Reviewer 2: > The answer depends on what your users would expect.

Wouldn't this be what their **customers** would expect?  I would think that for B2B the language would be more focused around our customer's customers, rather than the end user.

Not all users will trigger the logout process manually, so Auth0 also provides **session timeout** to prevent overly long-lived sessions. This setting is [available and configurable via the Auth0 Dashboard](/dashboard/reference/settings-tenant#login-session-management).

<% if (platform === "b2b") { %>

>  [Federated User Logout](/logout/guides/logout-idps) may be something that you need to consider for your application.  If you or your customers will be using a third-party IdP (i.e., something other than a [Database Identity Provider](/connections/database)) then the question of whether you need to log the user out of the IdP when they log out of your application is something you will need to answer. The answer depends on what your users would expect. If the application and/or IdP you use is tied closely to a customer organization and a central part of day-to-day operations, then it may be frustrating for users to get logged out of their IdP when they log out of your application. If not, then being logged out of the IdP may be expected, or in some cases even desired. In most B2B scenarios, our customers find that it is preferable *not* to perform federated logout for a user.

Author: This was moved to its own include file
